libsolidity/parsing/Parser.cpp

```diff
@@ -2199,11 +2201,26 @@ ASTPointer<ParameterList> Parser::createEmptyParameterList()
 
 ASTPointer<ASTString> Parser::expectIdentifierToken()
 {
-	// do not advance on success
-	expectToken(Token::Identifier, false);
+	expectToken(Token::Identifier, false /* do not advance */);
 	return getLiteralAndAdvance();
 }
 
+ASTPointer<ASTString> Parser::expectIdentifierTokenOrAddress()
+{
+	ASTPointer<ASTString> result;
+	if (m_scanner->currentToken() == Token::Address)
```

Comment:
Do we actually have any other such cases where it is valid as an identifier? Perhaps we'll need to extend this. Okay to merge this PR as is to limit the scope, but probably worth looking into this.

Reply:
Yes, we can extend it, but currently, nothing comes to my mind.